Mathlib/CategoryTheory/Generator/Basic.lean

```diff
@@ -60,143 +62,161 @@ namespace CategoryTheory
 
 variable {C : Type u₁} [Category.{v₁} C] {D : Type u₂} [Category.{v₂} D]
 
-/-- We say that `𝒢` is a separating set if the functors `C(G, -)` for `G ∈ 𝒢` are collectively
-    faithful, i.e., if `h ≫ f = h ≫ g` for all `h` with domain in `𝒢` implies `f = g`. -/
-def IsSeparating (𝒢 : Set C) : Prop :=
-  ∀ ⦃X Y : C⦄ (f g : X ⟶ Y), (∀ G ∈ 𝒢, ∀ (h : G ⟶ X), h ≫ f = h ≫ g) → f = g
-
-/-- We say that `𝒢` is a coseparating set if the functors `C(-, G)` for `G ∈ 𝒢` are collectively
-    faithful, i.e., if `f ≫ h = g ≫ h` for all `h` with codomain in `𝒢` implies `f = g`. -/
-def IsCoseparating (𝒢 : Set C) : Prop :=
-  ∀ ⦃X Y : C⦄ (f g : X ⟶ Y), (∀ G ∈ 𝒢, ∀ (h : Y ⟶ G), f ≫ h = g ≫ h) → f = g
-
-/-- We say that `𝒢` is a detecting set if the functors `C(G, -)` collectively reflect isomorphisms,
-    i.e., if any `h` with domain in `𝒢` uniquely factors through `f`, then `f` is an isomorphism. -/
-def IsDetecting (𝒢 : Set C) : Prop :=
-  ∀ ⦃X Y : C⦄ (f : X ⟶ Y), (∀ G ∈ 𝒢, ∀ (h : G ⟶ Y), ∃! h' : G ⟶ X, h' ≫ f = h) → IsIso f
-
-/-- We say that `𝒢` is a codetecting set if the functors `C(-, G)` collectively reflect
-    isomorphisms, i.e., if any `h` with codomain in `G` uniquely factors through `f`, then `f` is
-    an isomorphism. -/
-def IsCodetecting (𝒢 : Set C) : Prop :=
-  ∀ ⦃X Y : C⦄ (f : X ⟶ Y), (∀ G ∈ 𝒢, ∀ (h : X ⟶ G), ∃! h' : Y ⟶ G, f ≫ h' = h) → IsIso f
+namespace ObjectProperty
+
+variable (P : ObjectProperty C)
+
+/-- We say that `P : ObjectProperty C` is a separating set if the functors `C(G, -)`
```

Comment:
Very very minor nitpick, but is it still worth saying "separating set" in docstrings when they are now object properties (which are, admittedly, just an alias for sets)?

```suggestion
/-- We say that `P : ObjectProperty C` is separating if the functors `C(G, -)`
```

Same question for the docs in the few definitions below.